crates/re_components/src/load_file.rs

```diff
@@ -86,9 +104,24 @@ pub fn data_cells_from_file_contents(
         .to_string();
 
     match extension.as_str() {
-        "glb" => data_cells_from_mesh_file_contents(bytes, crate::MeshFormat::Glb),
-        "glft" => data_cells_from_mesh_file_contents(bytes, crate::MeshFormat::Gltf),
-        "obj" => data_cells_from_mesh_file_contents(bytes, crate::MeshFormat::Obj),
+        "glb" | "gltf" | "obj" => {
+            use re_types::{archetypes::Asset3D, components::MediaType, Archetype};
+            let cells: Result<Vec<_>, _> =
+                Asset3D::from_bytes(bytes, MediaType::guess_from_path(file_name))
+                    .as_component_batches()
+                    .into_iter()
+                    .map(|comp_batch| {
+                        let comp_batch = comp_batch.as_ref();
+                        Ok(DataCell::from_arrow(
+                            comp_batch.name(),
+                            comp_batch
+                                .try_to_arrow()
+                                .map_err(|err| anyhow::anyhow!("serialization failed: {err}"))?,
+                        ))
+                    })
+                    .collect();
```

Comment:
This code is identical – we could have a helper for this, e.g. in `trait Archetype`

Reply:
I wanted too, but that would require a dependency on `re_log_types` for `DataCell`, which is circular :(

I'll add a comment/issue.